pytorch_lightning/utilities/model_summary.py

```diff
@@ -450,14 +457,16 @@ def summarize(
 
     # temporary mapping from mode to max_depth
     if max_depth is None:
-        if mode in ModelSummaryMode.supported_types():
+        if mode is None:
+            model_summary = ModelSummary(lightning_module, max_depth=1)
+        elif mode in ModelSummaryMode.supported_types():
```

Comment:
reviewer: had to change the if-else this structure here. Otherwise we could have ended up with a reference to the variable `model_summary` before assignment